pip/basecommand.py

```diff
@@ -314,6 +314,23 @@ def populate_requirement_set(requirement_set, args, options, finder,
                     'You must give at least one requirement to %(name)s '
                     '(see "pip help %(name)s")' % opts)
 
+        # On Windows, any operation modifying pip should be run as:
+        #     python -m pip ...
+        # See https://github.com/pypa/pip/issues/1299 for more discussion
+        should_show_use_python_msg = (
+            sys.platform == 'win32' and
+            requirement_set.has_requirement('pip') and
+            "pip" in os.path.basename(sys.argv[0])
+        )
+        if should_show_use_python_msg:
+            new_command = [
+                sys.executable, "-m", "pip"
+            ] + sys.argv[1:]
+            raise CommandError(
+                'To modify pip, please run the following command:\n{}'
+                .format(" ".join(new_command))
+            )
```

Comment:
The whole point is to not proceed with a command that might modify the `pip` package. Otherwise, it still shows an unfriendly traceback which would bury this warning.



> That's arguably OK as we don't support using pip's API directly, but it's still a bit unfriendly...

I do think that one shouldn't go around breaking existing workflows in general. In this case though, I'd argue that the gains from breaking such tools (that already not supported) in exchange for a  _much_ better UX on running `pip install -U pip` on Windows is worth it.